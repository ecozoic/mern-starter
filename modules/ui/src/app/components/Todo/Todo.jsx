import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './Todo.scss';

/**
 * Renders a single todo
 * @param {Object} props - component props
 * @param {Function} props.onClick - function to call when todo is clicked
 * @param {boolean} props.completed - boolean indicating completion status of todo
 * @param {string} props.text - text describing todo
 */
const Todo = ({ onClick, completed, text }) => {
  const className = classNames(
    styles.todo,
    { [styles.completed]: completed },
  );

  return (
    <li
      className={className}
      onClick={onClick}
    >
      {text}
    </li>
  );
};

Todo.propTypes = {
  onClick: PropTypes.func.isRequired,
  completed: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
};

export default Todo;
