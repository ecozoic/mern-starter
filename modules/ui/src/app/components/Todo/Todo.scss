.todo {
  font-size: 18px;
  user-select: none;
}

.completed {
  text-decoration: line-through;
}
