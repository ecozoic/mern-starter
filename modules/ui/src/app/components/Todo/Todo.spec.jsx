import React from 'react';
import _ from 'lodash';
import { shallow } from 'enzyme';

import Todo from './';

describe('<Todo />', () => {
  it('renders without crashing', () => {
    const todo = shallow(
      <Todo
        onClick={_.identity}
        text="Todo"
        completed={false}
      />,
    );

    expect(todo).toMatchSnapshot();
  });

  it('displays todo text', () => {
    const todo = shallow(
      <Todo
        onClick={_.identity}
        text="Todo"
        completed={false}
      />,
    );

    const expected = 'Todo';
    const actual = todo.find('ListItem').render().text();

    expect(actual).toBe(expected);
  });

  it('has completed class if completed', () => {
    const todo = shallow(
      <Todo
        onClick={_.identity}
        text="Todo"
        completed
      />,
    );

    const classes = todo.find('ListItem').props().className.split(' ');

    expect(classes.indexOf('completed') > -1).toEqual(true);
  });

  it('does not have completed class if incomplete', () => {
    const todo = shallow(
      <Todo
        onClick={_.identity}
        text="Todo"
        completed={false}
      />,
    );

    const classes = todo.find('ListItem').props().className.split(' ');

    expect(classes.indexOf('completed') > -1).toEqual(false);
  });

  it('fires onclick when clicked', () => {
    const mockFn = jest.fn();

    const todo = shallow(
      <Todo
        onClick={mockFn}
        text="Todo"
        completed={false}
      />,
    );

    todo.find('ListItem').simulate('click');

    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});
