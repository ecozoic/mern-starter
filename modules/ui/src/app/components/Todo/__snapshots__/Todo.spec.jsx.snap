// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> renders without crashing 1`] = `
<li
  className="todo"
  onClick={[Function]}
>
  Todo
</li>
`;
